pkg/bmc/redfish_noop.go

```diff
@@ -0,0 +1,112 @@
+package bmc
+
+import (
+	"net/url"
+	"strings"
+)
+
+func init() {
+	schemes := []string{"http", "https"}
+	RegisterFactory("redfish-noop", newRedfishNoopAccessDetails, schemes)
+	RegisterFactory("ilo5-redfish-noop", newRedfishNoopAccessDetails, schemes)
+}
+
+func redfishNoopDetails(parsedURL *url.URL, disableCertificateVerification bool) *redfishAccessDetails {
+	return &redfishAccessDetails{
+		bmcType:                        parsedURL.Scheme,
+		host:                           parsedURL.Host,
+		path:                           parsedURL.Path,
+		disableCertificateVerification: disableCertificateVerification,
+	}
+}
+
+func newRedfishNoopAccessDetails(parsedURL *url.URL, disableCertificateVerification bool) (AccessDetails, error) {
+	return redfishDetails(parsedURL, disableCertificateVerification), nil
+}
+
+type redfishNoopAccessDetails struct {
+	bmcType                        string
+	host                           string
+	path                           string
+	disableCertificateVerification bool
+}
+
+const redfishNoopDefaultScheme = "https"
+
+func (a *redfishNoopAccessDetails) Type() string {
+	return a.bmcType
+}
+
+// NeedsMAC returns true when the host is going to need a separate
+// port created rather than having it discovered.
+func (a *redfishNoopAccessDetails) NeedsMAC() bool {
+	// For the inspection to work, we need a MAC address
+	// https://github.com/metal3-io/baremetal-operator/pull/284#discussion_r317579040
+	return true
+}
+
+func (a *redfishNoopAccessDetails) Driver() string {
+	return "redfish"
+}
+
+func (a *redfishNoopAccessDetails) DisableCertificateVerification() bool {
+	return a.disableCertificateVerification
+}
+
+func getRedfishNoopAddress(bmcType, host string) string {
```

Comment:
Could we move this to a common location? It's the same for all redfish implementations.